modules/core/ImagerySystem.js

```diff
@@ -398,6 +400,98 @@ export class ImagerySystem extends AbstractSystem {
   }
 
 
+  /**
+   *
+   */
+  async getWaybackSource() {
+    const context = this.context;
+    const view = context.systems.map.extent();
+    const urlhash = this.context.systems.urlhash;
+    const zoomLatLng = urlhash.getParam('map');
```

Comment:
There's nothing incorrect about your zoom/lat/lng calc code, but there's a better way to get this info- the map system! 

The map system is inited on line 70. You should be able to just do 

```
const zoom = map.zoom();
const latLng = {};
const [latLng.lon, latLng.lat] = map.loc(); 
```

Reply:
fixed with map.center(). so much cleaner!